src/Settings.pm

```diff
@@ -428,14 +428,26 @@ sub loadByHandle {
 	assert(defined $object) if DEBUG;
 
 	my $filename;
-	if ($object->{autoSearch}) {
-		if ($object->{type} == CONTROL_FILE_TYPE) {
-			$filename = _findFileFromFolders($object->{name}, \@controlFolders);
+	my $internalFilename;
+	if ($object->{'autoSearch'}) {
+		$internalFilename = $object->{'name'};
+	} else {
+		$internalFilename = $object->{'internalName'};
+	}
+
+	#hooks of type 'pre_load_' make it possible to change the filename before openkore searches for it (this filename must contain file name and file path)
+	my %pre_load = (internalFilename => $internalFilename, filename => \$filename);
+	Plugins::callHook('pre_load_'.$internalFilename, \%pre_load);
```

Comment:
Also it's a good time to add the proper verification for 'TABLE_FILE_TYPE' instead of assuming everything that isn't a 'CONTROL_FILE_TYPE' is a 'TABLE_FILE_TYPE', this adds support for new file types in the future.